pkg/operator/ceph/object/user/controller.go

```diff
@@ -334,6 +357,28 @@ func generateUserConfig(user *cephv1.CephObjectStoreUser) admin.User {
 		Keys:        make([]admin.UserKeySpec, 1),
 	}
 
+	if user.Spec.Quotas != nil && user.Spec.Quotas.MaxBuckets != 0 {
+		userConfig.MaxBuckets = &user.Spec.Quotas.MaxBuckets
+	}
+
+	if user.Spec.Capabilities != nil {
+		if user.Spec.Capabilities.User != "" {
+			userConfig.UserCaps += fmt.Sprintf("users=%s;", user.Spec.Capabilities.User)
+		}
+		if user.Spec.Capabilities.Bucket != "" {
+			userConfig.UserCaps += fmt.Sprintf("buckets=%s;", user.Spec.Capabilities.Bucket)
+		}
+		if user.Spec.Capabilities.MetaData != "" {
+			userConfig.UserCaps += fmt.Sprintf("metadata=%s;", user.Spec.Capabilities.MetaData)
+		}
+		if user.Spec.Capabilities.Usage != "" {
+			userConfig.UserCaps += fmt.Sprintf("usage=%s;", user.Spec.Capabilities.Usage)
+		}
+		if user.Spec.Capabilities.Zone != "" {
+			userConfig.UserCaps += fmt.Sprintf("zone=%s;", user.Spec.Capabilities.Zone)
+		}
+	}
```

Comment:
@BlaineEXE : Apparently, it looks like either bug in go-ceph or rgw(most likely) or updating caps is not working atm. It can be performed only while CreateUser API() not with ModifyUser(), even though the request is filled with `caps` details. I have mentioned the same in `ceph-object-user-crd.md`